src/libsyntax/attr.rs

```diff
@@ -317,6 +317,13 @@ pub fn find_export_name_attr(diag: &Handler, attrs: &[Attribute]) -> Option<Inte
     })
 }
 
+pub fn contains_extern_indicator(attrs: &[Attribute]) -> bool {
+    contains_name(attrs, "no_mangle") ||
+        contains_name(attrs, "link_section") ||
+        contains_name(attrs, "linkage") ||
```

Comment:
Also from #29676, the `linkage` and `link_section` attributes should be ignored here.